beacon_node/network/src/network_beacon_processor/sync_methods.rs

```diff
@@ -298,8 +296,27 @@ impl<T: BeaconChainTypes> NetworkBeaconProcessor<T> {
         });
     }
 
-    pub fn send_delayed_lookup(&self, block_root: Hash256) {
-        self.send_sync_message(SyncMessage::MissingGossipBlockComponentsDelayed(block_root))
+    /// Poll the beacon chain for any delayed lookups that are now available.
+    pub fn poll_delayed_lookups(&self, slot: Slot) {
+        let block_roots = self
+            .chain
+            .data_availability_checker
+            .incomplete_processing_components(slot);
+        if block_roots.is_empty() {
+            trace!(self.log, "No delayed lookups found on poll");
+        } else {
+            debug!(self.log, "Found delayed lookups on poll"; "lookup_count" => block_roots.len());
+        }
+        for block_root in block_roots {
+            if let Some(peer_ids) = self.delayed_lookup_peers.lock().pop(&block_root) {
+                // Let the network layer de-dupe these.
+                for peer_id in peer_ids {
+                    self.send_sync_message(SyncMessage::MissingGossipBlockComponents(
```

Comment:
Does it make sense to have `MissingGossipBlockComponents` take a `Vec<PeerId>` to avoid multiple channel messages?

Reply:
updated here: https://github.com/sigp/lighthouse/pull/4732/commits/75c50dbcef07eea7ea91fd4e91d38cdb9b74460c